File: tasks/2025/round4/maxdifference.py
#!/usr/bin/env python3
# insert brief description of the solution here
T = int(input())
for _ in range(T):
    N = int(input().strip())
    V = list(map(int, input().strip().split()))
    ans = 0
    setMax = -1e9
    setMin = -1e9
    for x in V:
        setMax = max(ans + x, setMax)
        setMin = max(ans - x, setMin)
        ans = max(setMax - x, setMin + x)

    print(ans)

File: tasks/2025/round4/maxdifference.cpp
#include <iostream>
using namespace std;
using ll = long long;

void solve(){
    int N; cin >> N;
    ll ans = 0;
    ll setMin = -1e9;
    ll setMax = -1e9;
    while(N--){
        ll x;
        cin >> x;
        setMax = max(setMax, ans + x);
        setMin = max(setMin, ans - x);
        ans = max(setMax - x, setMin + x);
    }

    cout << ans << endl;
}

int main() {
    ios_base::sync_with_stdio(0);
    int T;
    cin >> T;
    while(T--)solve();
}
